subprojects/core/src/integTest/groovy/org/gradle/process/internal/worker/DefaultWorkerProcessBuilderIntegrationTest.groovy

```diff
@@ -18,54 +18,78 @@ package org.gradle.process.internal.worker
 
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
-import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
-import org.gradle.integtests.fixtures.executer.IntegrationTestBuildContext
-
-import static org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache.Skip.INVESTIGATE
+import org.gradle.util.internal.TextUtil
 
 class DefaultWorkerProcessBuilderIntegrationTest extends AbstractIntegrationSpec {
 
-    @ToBeFixedForConfigurationCache(skip = INVESTIGATE)
     def "test classpath does not contain nonexistent entries"() {
-        String integTestHomeDir = IntegrationTestBuildContext.INSTANCE.getGradleUserHomeDir().absolutePath
         given:
-        file("src/test/java/ClasspathTest.java") << """
-        import org.junit.Test;
-
-        import static org.junit.Assert.assertTrue;
-
-        public class ClasspathTest {
-            @Test
-            public void test() {
-                String runtimeClasspath = System.getProperty("java.class.path");
-                System.out.println(runtimeClasspath);
-                assertTrue(runtimeClasspath.contains("${integTestHomeDir.replace("\\", "\\\\")}"));
-                assertTrue(!runtimeClasspath.contains("Non exist path"));
+        def existingDir = TextUtil.escapeString(createDir("existing").absolutePath)
+        def nonExistingDir = TextUtil.escapeString(new File(existingDir, "Non exist path").absolutePath)
+
```

Comment:
Thanks for fixing this test.

💅 

The existing and non-existing paths don't need to be related, right?

I would suggest storing the `TestFile`/`File` objects in local variables, and then have the escaped paths stored in an additional pair of vars. The call to `createDir` is a bit too easy to miss.
 
I also would assert in the test itself that the `existing` location exists, and the `nonExisting` location does not exist.

Finally, I would add an assertion that the test task did execute.